staging/src/k8s.io/cli-runtime/pkg/resource/selector.go

```diff
@@ -53,6 +55,14 @@ func (r *Selector) Visit(fn VisitorFunc) error {
 		FieldSelector: r.FieldSelector,
 		Limit:         r.LimitChunks,
 	}
+
+	if len(r.Namespace) > 0 {
```

Comment:
The current approach is inentional and we don't expect this to be changed.
/hold

Reply:
OK, thanks.